### PR TITLE
*: support new aggregation function var_pop and var_samp

### PR DESCRIPTION
**Pull request by mccxj**

### CODE REVIEW

**Comment by mccxj** [comment]
@zz-jason PTAL

**Comment by tiancaiamao** [comment]
Would you please fix the conflicts and link the TiDB PR. @mccxj 
For more instructions, see https://github.com/pingcap/parser/#how-to-update-parser-for-tidb

File: parser.y [hunk 1]
```diff
@@ -3853,13 +3857,37 @@ SumExpr:
 	{
 		$$ = &ast.AggregateFuncExpr{F: $1, Args: []ast.ExprNode{$4}, Distinct: $3.(bool)}
 	}
-|	builtinStddevPop '(' BuggyDefaultFalseDistinctOpt Expression ')'
+|	builtinStddevPop '(' Expression ')'
```
Comment (lzmhhh123) [inline parser.y]: We'd better support `distinct` clause in variance aggregate function.

Comment (zz-jason) [reply parser.y]: @mccxj After discussion, we decide to support `distinct` for every aggregate functions.

**Pull request closed without merging**